packages/ckeditor5-ui/src/autocomplete/autocompleteview.ts

```diff
@@ -0,0 +1,232 @@
+		this.listenTo( global.document, 'scroll', () => {
+			this._updateResultsViewWidthAndPosition();
+		} );
+
+		// Hide the results when the component becomes disabled.
+		this.on( 'change:isEnabled', () => {
+			this._updateResultsVisibility();
+		} );
+
+		// Update the value of the query field when the user selects a result.
+		this.filteredView.on<FilteredViewExecuteEvent>( 'execute', ( evt, { value } ) => {
+			// Focus the query view first to avoid losing the focus.
+			this.focus();
+
+			// Resetting the view will ensure that the #queryView will update its empty state correctly.
+			// This prevents bugs related to dynamic labels or auto-grow when re-setting the same value
+			// to #queryView.fieldView.value (which does not trigger empty state change) to an
+			// #queryView.fieldView.element that has been changed by the user.
+			this.reset();
+
+			// Update the value of the query field.
+			this.queryView.fieldView.value = this.queryView.fieldView.element!.value = value;
+
+			// Finally, hide the results view. The focus has been moved earlier so this is safe.
+			resultsView.isVisible = false;
+		} );
+
+		// Update the position and width of the results view when it becomes visible.
+		resultsView.on( 'change:isVisible', () => {
+			this._updateResultsViewWidthAndPosition();
+		} );
+	}
+
+	/**
+	 * Updates the position of the results view on demand.
+	 */
+	private _updateResultsViewWidthAndPosition() {
+		const resultsView = ( this.resultsView as AutocompleteResultsView );
+
+		if ( !resultsView.isVisible ) {
+			return;
+		}
+
+		resultsView._width = new Rect( this.queryView.fieldView.element! ).width;
+
+		const optimalResultsPosition = AutocompleteView._getOptimalPosition( {
+			element: this.resultsView.element!,
+			target: this.queryView.element!,
+			fitInViewport: true,
+			positions: AutocompleteView.defaultResultsPositions
+		} );
+
+		// _getOptimalPosition will return null if there is no optimal position found (e.g. target is off the viewport).
+		resultsView._position = optimalResultsPosition ? optimalResultsPosition.name as string : 's';
+	}
+
+	/**
+	 * Updates the visibility of the results view on demand.
+	 */
+	private _updateResultsVisibility() {
+		const resultsView = ( this.resultsView as AutocompleteResultsView );
+		const queryMinChars = typeof this._config.queryMinChars === 'undefined' ? 0 : this._config.queryMinChars;
+		const queryLength = this.queryView.fieldView.element!.value.length;
+
+		resultsView.isVisible = this.focusTracker.isFocused && this.isEnabled && queryLength >= queryMinChars;
+	}
+
+	/**
+	 * Positions for the autocomplete results view. Two positions are defined by default:
+	 * * `s` - below the search field,
+	 * * `n` - above the search field.
+	 */
+	public static defaultResultsPositions: Array<PositioningFunction> = [
+		( fieldRect => {
+			return {
+				top: fieldRect.bottom,
+				left: fieldRect.left,
+				name: 's'
+			};
+		} ) as PositioningFunction,
```

Comment:
```suggestion
		} ) satisfies PositioningFunction,
```
Supported since TS 4.9. It checks the function conforms to `PositioningFunction` (and makes contextual typing work), but doesn't _force_ it to `PositioningFunction`.

Reply:
The current version of Typescript is `"^4.8.4"`. My IDE doesn't work correctly with the `satisfies` and I'm not sure if it will compile properly. For now, I will stay with `as`.